presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -5607,6 +5608,84 @@ private void testSelectWithNoColumns(Session session, HiveStorageFormat storageF
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testColumnPruning()
```

Comment:
`testNestedColumnPrunning`?

Reply:
it's not only about nested columns (rows/structs). eg top-level `a` column is pruned too